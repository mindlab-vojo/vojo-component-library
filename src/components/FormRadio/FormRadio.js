import React,  { useState } from 'react'
import PropTypes from 'prop-types'

import ThemeConsumer from '../../style/ThemeManager/ThemeProvider'
import * as FormRadio from './FormRadio.style'
import IconManager from '../IconManager'
import Typography from '../Typography'

const FormRadioComponent = ({
  description,
  hasToolTip,
  onClickOption,
  onClickToolTip,
  options,
  title,
  tooltipData,
}) => {

  const defaultOption = {
    id: null,
    title: null,
    value: null
  }
  const [selectedOption, setSelectedOption] = useState(defaultOption)

  const handleClickOption = (option) => {
    setSelectedOption(option)
    onClickOption && onClickOption(option)
  }

  const handleClickToolTip = () => {
    onClickToolTip && onClickToolTip(tooltipData)
  }

  const renderOptions = (theme) => {
    return options.map(option => {
      const isSelected = option.id === selectedOption.id

      return (
        <FormRadio.Option
          key={option.id}
          onClick={() => handleClickOption(option)}>
          <FormRadio.HiddenRadio
            checked={isSelected}
            type="radio"
            value={option.value}
            id={option.id}
          />
          <FormRadio.Label
            htmlFor={option.id}>
            <FormRadio.RadioIcon>
              <IconManager 
                height="20px"
                width="20px"
                icon={
                  isSelected ?
                    "RadioChecked" : 
                    "RadioUnchecked"
                }
                fill={theme.colors.darkGreyColor}/>
            </FormRadio.RadioIcon>
            <Typography
              color={theme.colors.darkGreyColor}
              fontSize="14px">
              {option.title}
            </Typography>
          </FormRadio.Label>
        </FormRadio.Option>
      )
    })
  }

  const renderComponent = (theme) => (
    <FormRadio.Wrapper>
      <FormRadio.Header>
        <FormRadio.TitleWrapper>
          <FormRadio.Title>
            <Typography
              color={theme.colors.darkGreyColor}
              fontSize="16px"
              fontWeight="bold">
              {title}
            </Typography>
          </FormRadio.Title>
          <FormRadio.Description>
            <Typography
              color={theme.colors.darkGreyColor}
              fontSize="12px">
              {description}
            </Typography>
          </FormRadio.Description>
        </FormRadio.TitleWrapper>
        {
          hasToolTip &&(
            <FormRadio.Tooltip
              onClick={handleClickToolTip}>
              <IconManager 
                height="20px"
                width="20px"
                icon="HelpCircle"
                fill={theme.colors.darkGreyColor}/>
            </FormRadio.Tooltip>
          )
        }
      </FormRadio.Header>
      <FormRadio.OptionsWrapper>
        <FormRadio.OptionsWrapper>
          {
            renderOptions(theme)
          }
        </FormRadio.OptionsWrapper>
      </FormRadio.OptionsWrapper>
    </FormRadio.Wrapper>
  )

  return (
    <ThemeConsumer>
      {({ theme }) => renderComponent(theme)}
    </ThemeConsumer>
  )
}

FormRadioComponent.propTypes = {
  description: PropTypes.string,
  hasToolTip: PropTypes.bool,
  onClickOption: PropTypes.func,
  onClickToolTip: PropTypes.func,
  options: PropTypes.array,
  title: PropTypes.string,
  tooltipData: PropTypes.object,
}

export default FormRadioComponent
