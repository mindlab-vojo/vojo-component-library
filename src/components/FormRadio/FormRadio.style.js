import styled from 'styled-components'

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
`

export const Header = styled.div`
  display: flex;
`

export const TitleWrapper = styled.div`
  display: flex;
  flex-direction: column;
`

export const Title = styled.div`
  display: flex;
`

export const Tooltip = styled.div`
  display: flex;
`

export const Description = styled.div`
  display: flex;
`

export const OptionsWrapper = styled.div`
  display: flex;
  flex-direction: column;
`

export const Option = styled.div`
  display: flex;
  flex-direction: column;
`

export const HiddenRadio = styled.input`
`

export const RadioIcon = styled.div`
`

export const Label = styled.label`
`