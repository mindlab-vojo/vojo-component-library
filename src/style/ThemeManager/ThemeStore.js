import { vojo } from './../themes'

class ThemeStore {

  constructor(theme) {
    this.defaultTheme = vojo
    this.theme = theme ? theme : this.defaultTheme
  }

  configure = ({ theme }) =>  {
    if (theme) this.configureTheme(theme)
    else this.configureTheme(this.defaultTheme)
  }

  configureTheme = (theme) => {
    this.theme = theme
  }

  modify = (modifiedProps) => {
    const currentTheme = this.theme
    return { currentTheme, modifiedProps }
  }

  getTheme = () => this.theme

}

export default ThemeStore