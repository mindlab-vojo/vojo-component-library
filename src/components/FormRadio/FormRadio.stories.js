import React from 'react';
import { withKnobs, text, array, boolean, object } from '@storybook/addon-knobs';
import { withA11y } from '@storybook/addon-a11y';
import { action } from '@storybook/addon-actions'

import FormRadio from '.'; 
import { ThemeProvider } from "../../../src";
import categories from '../../../.storybook/categories';

export const formRadio = () => {
  const optionsList = [
    {
      id: '1',
      title: 'Opção 1',
      value: 'leonardo',
    },
    {
      id: '2',
      title: 'Opção 2',
      value: 'renata',
    },
    {
      id: '3',
      title: 'Opção 3',
      value: 'robson',
    },
    {
      id: '4',
      title: 'Opção 4',
      value: 'ana-carolina',
    },
  ]

  const tooltipDataObject = {
    title: "Título do Modal",
    description: "Descrição que aparecerá no modal, normalmente descreverá detalhes dos requisistos deste campo."
  }

  const errorMessage = text('errorMessage', 'Este campo possui algum erro...')
  const hasError = boolean('hasError', false)
  const hasToolTip = boolean('hasToolTip', true)
  const tooltipData = object('tooltipData', tooltipDataObject)
  const onClickOption = action('onClickOption')
  const onClickToolTip = action('onClickToolTip')
  const options = array('options', optionsList)
  const description = text('description', 'Breve descrição da pergunta')
  const title = text('title', 'Título da Pergunta')

  return (
    <FormRadio
      description={description}
      errorMessage={errorMessage}
      hasError={hasError}
      hasToolTip={hasToolTip}
      onClickOption={onClickOption}
      onClickToolTip={onClickToolTip}
      options={options}
      tooltipData={tooltipData}
      title={title}
    />
  )
}

export default {
  title: categories.Radios,
  component: formRadio,
  decorators: [
    withKnobs, 
    withA11y, 
    (story) => <ThemeProvider>{ story() }</ThemeProvider>,
    (story) => <div style={{ display: "flex" }}>{ story() }</div>,
  ],
}

formRadio.story = {
  name: 'FormRadio',
};
  