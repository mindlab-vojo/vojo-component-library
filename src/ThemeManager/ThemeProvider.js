import React, { Component, createContext } from 'react'
import PropTypes from 'prop-types'

import ThemeStore from './ThemeStore';

// Provider and Consumer instances
const { Provider, Consumer } = createContext();

class ThemeProvider extends Component {
  static propTypes = {
    children: PropTypes.node,
    theme: PropTypes.string
  }

  state = {
    themeConfig: new ThemeStore()
  }

  componentDidMount() {
    if (this.props.theme) this.setTheme()
  }

  componentDidUpdate(prevProps) {
    if (this.props.theme !== prevProps.theme) this.setTheme()
  }

  setTheme = () => {
    const { theme } = this.props
    const { themeConfig } = this.state
    const updatedTheme = themeConfig.configure({ theme })
    this.setState({ theme: updatedTheme })
  }

  render() {
    return (
      <Provider value={{ theme: this.state.themeConfig.theme }}>
        {this.props.children}
      </Provider>
    )
  }
}

export { ThemeProvider }

export default Consumer