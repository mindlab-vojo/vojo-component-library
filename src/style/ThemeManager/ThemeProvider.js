import React, { Component, createContext } from 'react'
import PropTypes from 'prop-types'

import ThemeStore from './ThemeStore';

// Provider and Consumer instances
const { Provider, Consumer } = createContext();

class ThemeProvider extends Component {
  static propTypes = {
    children: PropTypes.node,
    theme: PropTypes.string
  }

  state = {
    themeConfig: new ThemeStore()
  }

  componentDidMount() {
    if (this.props.theme) this.setTheme()
  }

  componentDidUpdate(prevProps) {
    if (this.props.theme !== prevProps.theme) this.setTheme()
  }

  setTheme = () => {
    const { theme } = this.props
    const { themeConfig } = this.state
    const updatedTheme = themeConfig.configure({ theme })
    this.setState({ theme: updatedTheme })
  }

  getThemeFont = () => {
    const { themeConfig } = this.state
    return themeConfig.font
  }

  render() {

    const GlobalFont = this.getThemeFont()

    return (
      <Provider value={{ theme: this.state.themeConfig.theme }}>
        <GlobalFont/>
        {this.props.children}
      </Provider>
    )
  }
}

export { ThemeProvider }

export default Consumer