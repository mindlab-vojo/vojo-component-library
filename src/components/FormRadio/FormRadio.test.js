import React from 'react';
import { render } from '@testing-library/react';

import FormRadio from './';
import { ThemeProvider } from './../../style/ThemeManager/ThemeProvider';

test('Test FormRadio component render', () => {
  const fakeFn = jest.fn()
  const component = render(
    <ThemeProvider>
      <FormRadio 
        description="Este é um exemplo de descrição"
        errorMessage="Este é um erro"
        hasError={true}
        hasToolTip={true}
        onClickOption={fakeFn}
        onClickToolTip={fakeFn}
        options={[
          {
            id: '1',
            title: 'Opção 1',
            value: 'leonardo',
          },
          {
            id: '2',
            title: 'Opção 2',
            value: 'renata',
          },
          {
            id: '3',
            title: 'Opção 3',
            value: 'robson',
          },
          {
            id: '4',
            title: 'Opção 4',
            value: 'ana-carolina',
          },
        ]}
        tooltipData={{
          title: "Título do Modal",
          description: "Descrição que aparecerá no modal, normalmente descreverá detalhes dos requisistos deste campo."
        }}
        title="Título"
      />
    </ThemeProvider>
  );
  expect(component).toMatchSnapshot()
});
