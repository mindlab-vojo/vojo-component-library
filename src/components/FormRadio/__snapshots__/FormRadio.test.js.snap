// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test FormRadio component render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-AxjAm bkUdLm"
      >
        <div
          class="sc-AxirZ eljDRB"
        >
          <div
            class="sc-AxiKw jWoTjM"
          >
            <div
              class="sc-AxhCb ZaNiK"
            >
              <p
                class="sc-fzpans jzxdbk"
              >
                Título
              </p>
            </div>
            <div
              class="sc-AxgMl bFfuHh"
            >
              <p
                class="sc-fzplWN gKsRiR"
              >
                Este é um exemplo de descrição
              </p>
            </div>
            <div
              class="sc-AxgMl bFfuHh"
            >
              <p
                class="sc-fznyAO irOkzZ"
              >
                Este é um erro
              </p>
            </div>
          </div>
          <div
            class="sc-AxhUy cWBCut"
          >
            <svg
              height="20px"
              viewBox="0 0 24 24"
              width="20px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 17h-2v-2h2v2zm2.07-7.75l-.9.92C13.45 12.9 13 13.5 13 15h-2v-.5c0-1.1.45-2.1 1.17-2.83l1.24-1.26c.37-.36.59-.86.59-1.41 0-1.1-.9-2-2-2s-2 .9-2 2H8c0-2.21 1.79-4 4-4s4 1.79 4 4c0 .88-.36 1.68-.93 2.25z"
                fill="4D4771"
              />
            </svg>
          </div>
        </div>
        <div
          class="sc-AxheI ghZMUJ"
        >
          <div
            class="sc-AxheI ghZMUJ"
          >
            <div
              class="sc-Axmtr dLmrMf"
            >
              <input
                class="sc-AxmLO hemNFY"
                id="1"
                type="radio"
                value="leonardo"
              />
              <label
                class="sc-fzoLsD ijPeEE"
                for="1"
              >
                <div
                  class="sc-fzozJi gQFIXz"
                >
                  <svg
                    height="20px"
                    viewBox="0 0 24 24"
                    width="20px"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                      fill="4D4771"
                    />
                  </svg>
                </div>
                <p
                  class="sc-fznKkj gxMqNm"
                >
                  Opção 1
                </p>
              </label>
            </div>
            <div
              class="sc-Axmtr dLmrMf"
            >
              <input
                class="sc-AxmLO hemNFY"
                id="2"
                type="radio"
                value="renata"
              />
              <label
                class="sc-fzoLsD ijPeEE"
                for="2"
              >
                <div
                  class="sc-fzozJi gQFIXz"
                >
                  <svg
                    height="20px"
                    viewBox="0 0 24 24"
                    width="20px"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                      fill="4D4771"
                    />
                  </svg>
                </div>
                <p
                  class="sc-fznZeY gkYovW"
                >
                  Opção 2
                </p>
              </label>
            </div>
            <div
              class="sc-Axmtr dLmrMf"
            >
              <input
                class="sc-AxmLO hemNFY"
                id="3"
                type="radio"
                value="robson"
              />
              <label
                class="sc-fzoLsD ijPeEE"
                for="3"
              >
                <div
                  class="sc-fzozJi gQFIXz"
                >
                  <svg
                    height="20px"
                    viewBox="0 0 24 24"
                    width="20px"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                      fill="4D4771"
                    />
                  </svg>
                </div>
                <p
                  class="sc-fzokOt okGQB"
                >
                  Opção 3
                </p>
              </label>
            </div>
            <div
              class="sc-Axmtr dLmrMf"
            >
              <input
                class="sc-AxmLO hemNFY"
                id="4"
                type="radio"
                value="ana-carolina"
              />
              <label
                class="sc-fzoLsD ijPeEE"
                for="4"
              >
                <div
                  class="sc-fzozJi gQFIXz"
                >
                  <svg
                    height="20px"
                    viewBox="0 0 24 24"
                    width="20px"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                      fill="4D4771"
                    />
                  </svg>
                </div>
                <p
                  class="sc-fzqBZW cleajo"
                >
                  Opção 4
                </p>
              </label>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-AxjAm bkUdLm"
    >
      <div
        class="sc-AxirZ eljDRB"
      >
        <div
          class="sc-AxiKw jWoTjM"
        >
          <div
            class="sc-AxhCb ZaNiK"
          >
            <p
              class="sc-fzpans jzxdbk"
            >
              Título
            </p>
          </div>
          <div
            class="sc-AxgMl bFfuHh"
          >
            <p
              class="sc-fzplWN gKsRiR"
            >
              Este é um exemplo de descrição
            </p>
          </div>
          <div
            class="sc-AxgMl bFfuHh"
          >
            <p
              class="sc-fznyAO irOkzZ"
            >
              Este é um erro
            </p>
          </div>
        </div>
        <div
          class="sc-AxhUy cWBCut"
        >
          <svg
            height="20px"
            viewBox="0 0 24 24"
            width="20px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 17h-2v-2h2v2zm2.07-7.75l-.9.92C13.45 12.9 13 13.5 13 15h-2v-.5c0-1.1.45-2.1 1.17-2.83l1.24-1.26c.37-.36.59-.86.59-1.41 0-1.1-.9-2-2-2s-2 .9-2 2H8c0-2.21 1.79-4 4-4s4 1.79 4 4c0 .88-.36 1.68-.93 2.25z"
              fill="4D4771"
            />
          </svg>
        </div>
      </div>
      <div
        class="sc-AxheI ghZMUJ"
      >
        <div
          class="sc-AxheI ghZMUJ"
        >
          <div
            class="sc-Axmtr dLmrMf"
          >
            <input
              class="sc-AxmLO hemNFY"
              id="1"
              type="radio"
              value="leonardo"
            />
            <label
              class="sc-fzoLsD ijPeEE"
              for="1"
            >
              <div
                class="sc-fzozJi gQFIXz"
              >
                <svg
                  height="20px"
                  viewBox="0 0 24 24"
                  width="20px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    fill="4D4771"
                  />
                </svg>
              </div>
              <p
                class="sc-fznKkj gxMqNm"
              >
                Opção 1
              </p>
            </label>
          </div>
          <div
            class="sc-Axmtr dLmrMf"
          >
            <input
              class="sc-AxmLO hemNFY"
              id="2"
              type="radio"
              value="renata"
            />
            <label
              class="sc-fzoLsD ijPeEE"
              for="2"
            >
              <div
                class="sc-fzozJi gQFIXz"
              >
                <svg
                  height="20px"
                  viewBox="0 0 24 24"
                  width="20px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    fill="4D4771"
                  />
                </svg>
              </div>
              <p
                class="sc-fznZeY gkYovW"
              >
                Opção 2
              </p>
            </label>
          </div>
          <div
            class="sc-Axmtr dLmrMf"
          >
            <input
              class="sc-AxmLO hemNFY"
              id="3"
              type="radio"
              value="robson"
            />
            <label
              class="sc-fzoLsD ijPeEE"
              for="3"
            >
              <div
                class="sc-fzozJi gQFIXz"
              >
                <svg
                  height="20px"
                  viewBox="0 0 24 24"
                  width="20px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    fill="4D4771"
                  />
                </svg>
              </div>
              <p
                class="sc-fzokOt okGQB"
              >
                Opção 3
              </p>
            </label>
          </div>
          <div
            class="sc-Axmtr dLmrMf"
          >
            <input
              class="sc-AxmLO hemNFY"
              id="4"
              type="radio"
              value="ana-carolina"
            />
            <label
              class="sc-fzoLsD ijPeEE"
              for="4"
            >
              <div
                class="sc-fzozJi gQFIXz"
              >
                <svg
                  height="20px"
                  viewBox="0 0 24 24"
                  width="20px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    fill="4D4771"
                  />
                </svg>
              </div>
              <p
                class="sc-fzqBZW cleajo"
              >
                Opção 4
              </p>
            </label>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
